sriov/main.go

```diff
@@ -56,9 +57,18 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 
 	if n.DeviceInfo != nil && n.DeviceInfo.PCIaddr != "" && n.DeviceInfo.Vfid >= 0 && n.DeviceInfo.Pfname != "" {
-		if err = setupVF(n, args.IfName, args.ContainerID, netns); err != nil {
-			return fmt.Errorf("failed to set up pod interface %q from the device %q: %v", args.IfName, n.Master, err)
-		}
+		err = setupVF(n, args.IfName, args.ContainerID, netns)
+		defer func() {
+			if err != nil {
+				err = netns.Do(func(_ ns.NetNS) error {
+					_, err := netlink.LinkByName(args.IfName)
```

Comment:
In dpdk mode LinkByName(args.IfName) will fail in container and we still need to releaseVF with dpdk mode.

Reply:
It might be good to split the setup of VF kernel interface and VF dpdk interface to separate calls, then we can have different path here to release VF. And even further, since we support both dpdk and sriov in sriov-cni and now there is a pkg/dpdk folder for dpdk related handlers, soon we will have another folder pkg/sriov for sriov related handlers, will it be applicable to have existing functions separated for both types in current code framework? I understand that there are willingness to re-use as much code base as possible.

Reply:
The releaseVF() takes care of interface in kernel mode or dpdk mode. This defer block will work with VF attached to kernel driver. VF with dpdk driver, this block won't call releaseVF should setupVF above fails for whatever reason. The solution here I can think of: 

- Check for dpdk mode
- skip call to releaseVF(i.e., return from here) only if _!DPDKMode && ( interface is not found in container NS)_ [ the _!DPDKMode_ here is Short-circuit evaluation; so checking interface will happen only if its not in dpdk mode]
- else call releaseVF()

Reply:
if DPDK is enabled then releaseVF will solve it, I added the code to return the error that occurred in setupVF

Reply:
Indeed releaseVF solves this but what I'm saying is that  releaseVF will not be called when L:64 returns error because VF is with dpdk driver. We do want to call releaseVF regardless of VF in dpdk mode or not.  The check on L:67 will prevent this.